Firestore/core/src/core/key_field_in_filter.h

```diff
@@ -32,15 +34,15 @@ namespace core {
  */
 class KeyFieldInFilter : public FieldFilter {
  public:
-  KeyFieldInFilter(model::FieldPath field, model::FieldValue value);
+  /** Creates a new document keys filter. Takes ownership of `value`. */
+  KeyFieldInFilter(const model::FieldPath& field,
+                   google_firestore_v1_Value value);
 
  private:
   class Rep;
 
-  static bool Contains(const model::FieldValue::Array& array_value,
-                       const model::Document& doc);
-
-  static void ValidateArrayValue(const model::FieldValue& value);
+  static std::set<model::DocumentKey> ExtractDocumentKeysFromValue(
```

Comment:
`std::set` generally has poor performance characteristics because it uses a linked structure internally that is not cache friendly on modern CPUs. Is it possible to use `std::unordered_set` instead? Also in `key_field_not_in_filter.h`.